src/pages/index.js

```diff
@@ -21,8 +21,71 @@ import {
   config,
   imagePreview,
   imgPreviewTitle,
+  profileAvatarButton,
+  submitDeleteButton,
+  cardSelector,
+  editAvatarForm,
 } from "../utils/constants.js";
 
+import PopupWithCardDelete from "../components/PopupWithCardDelete.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b316ef3e-b4c0-4fe5-a1ba-6f1194cf61a8",
+    "Content-Type": "application/json",
+  },
+});
+let cardSection;
+let userId;
+
+const userInfo = new UserInfo({
+  userNameSelector: ".profile__name",
+  userDescriptionSelector: ".profile__description",
+  userProfileSelector: ".profile__image",
+});
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
+    userId = userData._id;
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setAvatarInfo(userData.avatar);
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (data) => {
+          const newCard = renderCard(data);
+          cardSection.addItem(newCard);
+        },
+      },
+      cardsWrap
+    );
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+const avatarInformation = new PopupWithForm(
+  "#modal-avatar",
+  handleAvatarImage
+);
+avatarInformation.setEventListeners();
+
+const profileEditPopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+
+const avatarFormEl = document.querySelector(`#modal-avatar`);
+const avatarValidator = new FormValidator(config, avatarFormEl);

```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```